### PR TITLE
New Resource: aws_iot_thing

### PR DESCRIPTION
**Pull request by radeksimko**
```
TF_ACC=1 go test ./aws -v -run=TestAccAWSIotThing_ -timeout 120m
=== RUN   TestAccAWSIotThing_basic
--- PASS: TestAccAWSIotThing_basic (59.38s)
=== RUN   TestAccAWSIotThing_full
--- PASS: TestAccAWSIotThing_full (377.55s)
=== RUN   TestAccAWSIotThing_importBasic
--- PASS: TestAccAWSIotThing_importBasic (33.18s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	470.151s
```

### CODE REVIEW

File: aws/resource_aws_iot_thing_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		if rs.Type != "aws_iot_thing" {
+			continue
+		}
+
+		params := &iot.DescribeThingInput{
+			ThingName: aws.String(rs.Primary.ID),
+		}
+
+		_, err := conn.DescribeThing(params)
+		if err == nil {
```
Comment (bflad) [inline aws/resource_aws_iot_thing_test.go]: Should we flip this around to return `nil` on `iot.ErrCodeResourceNotFoundException` error and return the actual error on others (e.g. permission errors)?

File: website/docs/r/iot_thing.html.markdown [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
+layout: "aws"
+page_title: "AWS: aws_iot_thing"
+sidebar_current: "docs-aws-resource-iot-thing"
+description: |-
+    Creates and manages an AWS IoT Thing.
+---
+
+# aws_iot_thing
```
Comment (bflad) [inline website/docs/r/iot_thing.html.markdown]: Missing `## Import` documentation

File: aws/resource_aws_iot_thing.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+func resourceAwsIotThing() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsIotThingCreate,
+		Read:   resourceAwsIotThingRead,
+		Update: resourceAwsIotThingUpdate,
+		Delete: resourceAwsIotThingDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```
[the remark this answers is not in the document]
Comment (radeksimko) [reply aws/resource_aws_iot_thing.go]: Yeah, that's a piece of code I somehow forgot here as I was refactoring the CRUD.

**Comment by radeksimko** [comment]
@bflad PTAL, I believe I addressed all your comments 😉 

```
=== RUN   TestAccAWSIotThing_basic
--- PASS: TestAccAWSIotThing_basic (18.44s)
=== RUN   TestAccAWSIotThing_full
--- PASS: TestAccAWSIotThing_full (351.33s)
=== RUN   TestAccAWSIotThing_importBasic
--- PASS: TestAccAWSIotThing_importBasic (19.77s)
```

File: aws/resource_aws_iot_thing_test.go [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+		if rs.Type != "aws_iot_thing" {
+			continue
+		}
+
+		params := &iot.DescribeThingInput{
+			ThingName: aws.String(rs.Primary.ID),
+		}
+
+		_, err := conn.DescribeThing(params)
+		if err != nil {
```
Comment (bflad) [inline aws/resource_aws_iot_thing_test.go]: Aw shucks - just missing a `return err` in here, otherwise we print out the wrong message below

**Comment by bflad** [comment]
This has been released in [version 1.11.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1110-march-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**